src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -554,7 +554,8 @@ def do_compile(self,
       raise TaskError("Compilation failure: {}".format(e))
 
   def _record_compile_classpath(self, classpath, targets, outdir):
-    text = '\n'.join(classpath)
+    relative_classpaths = [os.path.relpath(path) for path in classpath]
```

Comment:
I think this should generate relpaths relative to the buildroot explicitly, using the value from `get_buildroot()`. That way if the working directory for compile changes, the relative paths will still be relative to the build root.